cli/cli/helpers/portal_manager/binary_artifact_getter.go

```diff
@@ -54,48 +56,60 @@ func DownloadLatestKurtosisPortalBinary(ctx context.Context) (bool, error) {
+	var release *github.RepositoryRelease
+	if portalTag == portalTagLatest {
+		release, err = getLatestRelease(ctx, ghClient)
+	} else {
+		release, err = getReleaseByTag(ctx, ghClient, portalTag)
+	}
 	if err != nil {
 		return false, defaultToCurrentVersionOrError(currentVersionStrMaybe, err)
 	}
 
-	latestVersionStr, currentVersionMatchesLatest, err := compareLatestVersionWithCurrent(currentVersionStrMaybe, latestRelease)
+	requiredVersionStr, currentVersionMatchesRequired, err := compareRequiredVersionWithCurrent(currentVersionStrMaybe, release)
 	if err != nil {
 		return false, defaultToCurrentVersionOrError(currentVersionStrMaybe, err)
 	}
-	if currentVersionMatchesLatest {
-		logrus.Infof("Kurtosis Portal version '%s' is the latest and already installed", latestVersionStr)
+	if currentVersionMatchesRequired {
+		logrus.Infof("Installed Kurtosis Portal version '%s' is the required one", requiredVersionStr)
 		return false, nil
 	}
 
 	if currentVersionStrMaybe == "" {
 		// no Portal is currently installed. Installing a brand new one
-		logrus.Infof("Installing Kurtosis Portal version '%s'", latestVersionStr)
+		logrus.Infof("Installing Kurtosis Portal version '%s'", requiredVersionStr)
 	} else {
-		logrus.Infof("Upgrading currently Kurtosis Portal version '%s' to '%s'", currentVersionStrMaybe, latestVersionStr)
+		logrus.Infof("Upgrading currently Kurtosis Portal version '%s' to '%s'", currentVersionStrMaybe, requiredVersionStr)
 	}
 
-	githubAssetContent, err := downloadGithubAsset(ctx, ghClient, latestRelease)
+	githubAssetContent, err := downloadGithubAsset(ctx, ghClient, release)
 	if err != nil {
 		return false, defaultToCurrentVersionOrError(currentVersionStrMaybe, err)
 	}
 
-	if err = extractAssetTgzToBinaryFileOnDisk(githubAssetContent, latestVersionStr, binaryFilePath, currentVersionFilePath); err != nil {
+	if err = extractAssetTgzToBinaryFileOnDisk(githubAssetContent, requiredVersionStr, binaryFilePath, currentVersionFilePath); err != nil {
 		return false, defaultToCurrentVersionOrError(currentVersionStrMaybe, err)
 	}
 	return true, nil
 }
 
 func getLatestRelease(ctx context.Context, ghClient *github.Client) (*github.RepositoryRelease, error) {
-	// First, browse the list of releases for the Kurtosis Portal repo
 	latestRelease, _, err := ghClient.Repositories.GetLatestRelease(ctx, kurtosisTechGithubOrg, kurtosisPortalGithubRepoName)
 	if err != nil {
-		return nil, stacktrace.Propagate(err, "Unable to retrieve latest version of Kurtosis Portal form GitHub")
+		return nil, stacktrace.Propagate(err, "Unable to retrieve latest version of Kurtosis Portal from GitHub")
 	}
 	logrus.Debugf("Latest release is %s", latestRelease.GetName())
 	return latestRelease, nil
 }
 
+func getReleaseByTag(ctx context.Context, ghClient *github.Client, tag string) (*github.RepositoryRelease, error) {
+	release, _, err := ghClient.Repositories.GetReleaseByTag(ctx, kurtosisTechGithubOrg, kurtosisPortalGithubRepoName, tag)
```

Comment:
Yes I'd basically move your `if == "latest"` in here either call `GetLatestRelease` or `GetReleaseByTag`. This way you have only one function, and you move a bit of the complexity inside this simple function, rather than adding some to the already "fairly complex" upstream one.